src/AstGen.zig

```diff
@@ -1884,15 +1922,22 @@ fn breakExpr(parent_gz: *GenZir, parent_scope: *Scope, node: Ast.Node.Index) Inn
                 }
                 block_gz.break_count += 1;
 
-                // The loop scope has a mechanism to prevent rvalue() from emitting a
-                // store to the result location for the loop body (since it is continues
-                // rather than returning a result from the loop) but here is a `break`
-                // which needs to override this behavior.
-                const prev_rvalue_noresult = parent_gz.rvalue_noresult;
-                parent_gz.rvalue_noresult = .none;
+                // unwrap if break_result_loc is only_break
+                const prev_break_result_loc = block_gz.break_result_loc;
+                const prev_rl_ptr = block_gz.rl_ptr;
+                const is_only_break = prev_break_result_loc == .only_break;
+                if (is_only_break) {
+                    block_gz.setBreakResultLoc(prev_break_result_loc.only_break.unwrap());
+                }
+                defer {
+                    if (is_only_break) {
+                        block_gz.setBreakResultLoc(prev_break_result_loc);
+                        block_gz.rl_ptr = prev_rl_ptr;
+                    }
+                }
```

Comment:
I think modifying the block's break result loc might lead to incorrect behavior with nested breaks like:
```zig
break :loop {
    if (cond) break :loop something;
}
```
Assigning `block_gz.break_result_loc` to a variable and then unwrapping if needed should avoid any issues.